integration/tests/basic/rpc/test_rpc_estimate_gas.py

```diff
@@ -0,0 +1,173 @@
+        assert estimated_gas >= int(
+            deploy_trx_big_gas["gasUsed"]
+        ), "Estimated Gas < Used Gas"
+
+    def test_rpc_estimate_gas_send_neon(self):
+        tx_receipt = self.send_neon(self.sender_account, self.recipient_account, amount=0.001)
+        transaction = self.web3_client.get_transaction_by_hash(tx_receipt["transactionHash"])
+
+        assert "gas" in transaction
+        estimate_gas = transaction["gas"]
+        assert estimate_gas == 30_000
+
+    def test_rpc_estimate_gas_erc20(self, erc20_simple):
+        tx_receipt = erc20_simple.transfer(erc20_simple.owner, self.recipient_account, 1)
+        transaction = self.web3_client.get_transaction_by_hash(tx_receipt["transactionHash"])
+
+        assert "gas" in transaction
+        estimate_gas = transaction["gas"]
+        assert estimate_gas == 1_552_280
+
+    def test_rpc_estimate_gas_spl(self, erc20_spl):
+        tx_receipt = erc20_spl.transfer(erc20_spl.account, self.recipient_account, 1)
+        transaction = self.web3_client.get_transaction_by_hash(tx_receipt["transactionHash"])
+
+        assert "gas" in transaction
+        estimate_gas = transaction["gas"]
+        assert estimate_gas == 2_084_280
+
+    def test_rpc_estimate_gas_contract_get_value(self, common_contract):
+        tx = self.make_contract_tx_object()
+        instruction_tx = common_contract.functions.getText().build_transaction(tx)
+        tx_receipt = self.web3_client.send_transaction(self.sender_account, instruction_tx)
+        transaction = self.web3_client.get_transaction_by_hash(tx_receipt["transactionHash"])
+
+        assert "gas" in transaction
+        estimate_gas = transaction["gas"]
+        assert estimate_gas == 30_000
+
+    def test_rpc_estimate_gas_contract_set_value(self, common_contract):
+        tx = self.make_contract_tx_object()
+        instruction_tx = common_contract.functions.setNumber(100).build_transaction(tx)
+        tx_receipt = self.web3_client.send_transaction(self.sender_account, instruction_tx)
+        transaction = self.web3_client.get_transaction_by_hash(tx_receipt["transactionHash"])
+
+        assert "gas" in transaction
+        estimate_gas = transaction["gas"]
+        assert estimate_gas == 30_000
+
+    def test_rpc_estimate_gas_contract_calls_another_contract(self, common_contract):
+        caller_contract, _ = self.web3_client.deploy_and_get_contract(
+            "common/Common", "0.8.12",
+            contract_name="CommonCaller", account=self.sender_account,
+            constructor_args=[common_contract.address]
+        )
+        tx = self.make_contract_tx_object()
+        instruction_tx = caller_contract.functions.getNumber().build_transaction(tx)
+        tx_receipt = self.web3_client.send_transaction(self.sender_account, instruction_tx)
+
+        transaction = self.web3_client.get_transaction_by_hash(tx_receipt["transactionHash"])
+
+        assert "gas" in transaction
+        estimate_gas = transaction["gas"]
```

Comment:
eth_estimateGas has additional argument block but you didn't check it https://docs.alchemy.com/reference/eth-estimategas

Reply:
Added